aldryn_forms/cms_plugins.py

```diff
@@ -398,6 +424,76 @@ def form_post_save(self, instance, form):
             self.send_notification_email(email, form, instance)
 
 
+class FileField(Field):
+    name = _('File upload field')
+
+    model = models.FileUploadFieldPlugin
+
+    form = FileFieldForm
+    form_field = RestrictedFileField
+    form_field_widget = RestrictedFileField.widget
+    form_field_enabled_options = [
+        'label',
+        'help_text',
+        'required',
+        'error_messages',
+        'validators',
+    ]
+    fieldset_general_fields = Field.fieldset_general_fields + [
+        'upload_to',
+        'max_size',
+    ]
+
+    def get_form_field_kwargs(self, instance):
+        kwargs = super(FileField, self).get_form_field_kwargs(instance)
+        if instance.max_size:
+            if 'help_text' in kwargs:
+                kwargs['help_text'] = kwargs['help_text'].replace(
+                    'MAXSIZE', filesizeformat(instance.max_size))
+            kwargs['max_size'] = instance.max_size
+        return kwargs
+
+    def serialize_value(self, instance, value):
+        return value.absolute_uri if value else '-'
+
+    def form_pre_save(self, instance, form, request, **kwargs):
+        """Save the uploaded file to django-filer
+
+        The type of model (file or image) is automatically chosen by trying to
+        open the uploaded file.
+        """
+        field_name = self.get_field_name(instance)
+        uploaded_file = form.cleaned_data[field_name]
+
+        if uploaded_file is None:
+            return
+
+        try:
+            with Image.open(uploaded_file) as im:
```

Comment:
The context is small enough for it to be readable anyway... I'll concede you an extra `g` :-P